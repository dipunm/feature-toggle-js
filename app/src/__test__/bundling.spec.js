const path = require('path');
const fs = require('fs');
const cmd = require('node-cmd');
const rmrf = require('rimraf');

const PATH_PACKAGES = path.join(__dirname, '../../../__packages__/');
const uniqueTextFromValidationLibrary = 'Must be invoked on a Joi instance';

describe('webpack bundling', () => {
  const workingDir = path.join(PATH_PACKAGES, 'webpack-bundle');

  beforeEach((done) => {
    rmrf(path.join(workingDir, 'dist'), (rmrfErr) => {
      if (rmrfErr) {
        done(rmrfErr);
      } else {
        done();
      }
    });
  });

  describe('for the browser', () => {
    let output;
    beforeEach((done) => {
      cmd.get(
        `(cd ${workingDir} && exec npm run build-browser)`,
        (buildErr, data) => {
          if (buildErr) {
            done({ data, buildErr });
          }
          output = fs.readFileSync(
            path.join(workingDir, 'dist', 'bundle.js'),
            'utf8',
          );
          done();
        },
      );
    }, 20000);

    test('should not include the joi validation library', () => {
      expect(output).not.toContain(uniqueTextFromValidationLibrary);
    });
  });

  describe('for node', () => {
    let output;
    beforeEach((done) => {
      cmd.get(
        `(cd ${workingDir} && exec npm run build-node)`,
        (buildErr, data) => {
          if (buildErr) {
            done({ data, buildErr });
          }
          output = fs.readFileSync(
            path.join(workingDir, 'dist', 'bundle.js'),
            'utf8',
          );
          done();
        },
      );
    }, 20000);

    test('should include the joi validation library', () => {
      expect(output).toContain(uniqueTextFromValidationLibrary);
    });
  });
});
