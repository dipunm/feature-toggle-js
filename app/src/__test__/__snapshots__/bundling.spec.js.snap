// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack bundling for the browser should execute with the expected output 1`] = `
"toggle1:  true
toggle2:  false
"
`;

exports[`webpack bundling for the browser with the lite client should execute with the expected output 1`] = `
"toggle1:  true
toggle2:  false
"
`;
